doc/HLD_Data_Block_Allocator.rst

```diff
@@ -0,0 +1,407 @@
+=====================
+Data Block Allocator
+=====================
+
+This document presents a high level design (HLD) of a data block allocator for Motr C2 core. The main purposes of this document are: (i) to be inspected by C2 architects and peer designers to ascertain that high level design is aligned with C2 architecture and other designs, and contains no defects, (ii) to be a source of material for Active Reviews of Intermediate Design (ARID) and detailed level design (DLD) of the same component, (iii) to serve as a design reference document.
```

Comment:
Remove the "C2".
If "C2" appears alone, replace it with Motr.

Reply:
Incorporated.